### ISSUE TITLE
Check usability on NFS filesystem (and or GPFS?)

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/disk-objectstore#30]
Apart from typical local FS, I expect that users might want to use on network file systems, that might not be able to provide certain guarantees like the atomicity of certain operations.

I am anyway setting this as "nice to have" because (I think that) with non-concurrent packing, the risks of problems are relatively low. This should be instead addressed as soon as we have fixed #4.

**Comment by ramirezfranciscof** [comment]
Here is a summary of some tests I performed to check the behaviour in GPFS systems. We still need to check how to repeat this tests in NFS to have a more complete picture.

**TESTS PERFORMED:**
First I check that all the pytests in the `test` folder run with no problem, then I run `test/test_benchmark.py` and `disk-objectstore/examples/example_objectstore.py` a couple of times (3-5) to check consistency of the results. To try different partitions I will change the `dirpath = tempfile.mkdtemp()` line in `tests/conftest.py` for the first one, and use the `-p path/to/dir` for the second one. Also, in the second case I will run with `-n 10000` for sufficient time sampling, as well as with/without `-d` to test the case of directly writing to packs.

**SYSTEMS TESTED:**
I have tried running these tests in my own desktop at EPFL (both in my `/home` and in `/tmp`, because they are in different drives), as well as in the EPFL cluster fidis (only in my `/home`) and in CSCS cluster Piz Daint (both in my `/home` and `/scratch`, which seem to be in different kinds of partitions).

**SUMMARY OF RESULTS:**
- All benchmarks and tests seemed to be around the same order of magnitude. This also includes the results for the `/home` and `/tmp` partitions in my theospc (noting that the first resides in a magnetic drive and a second in a solid state one), where the only noticeable improvement was when writing loose objects (`example_objectstore.py`).
- Inside of Daint: the `/home` partition seemed twice as fast as the `/scratch` one when running `example_objectstore.py`. In most of the `test_benchmark.py` they seemed more similar, except for `test_loose_read` (`/home` x3 faster) and `test_loose_write` (`/home` x7 faster).
- Also on Daint: the `max` values in the `test_list_all_packed` and `test_list_all_loose` entries of `test_benchmark.py` were a bit too variable, changing up to more than one order of magnitude between different runs. These are relatively small values so it may not be so significant (although there are other small time values that don't change as much).
- Comparing the `/home` of Daint and my theospc, the second one is around 3 times faster than the first. Fidis times are similar to Daint (except maybe in one or two cases, like when writing loose objects in `example_objectstore.py`, where the results where more similar to the ones of my theospc).
